base/install_area_detector.sh

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env bash
+
+set -ex
+
+git clone --depth 1 --branch ${AREA_DETECTOR_VERSION} \
+    https://github.com/areaDetector/areaDetector
+
+cd areaDetector
+
+git submodule update --init --depth 1 -j ${JOBS} \
+    ADSupport \
+    ADCore
+
+cd configure
+
+module_releases="
+AREA_DETECTOR=${EPICS_MODULES_PATH}/areaDetector
+ADSUPPORT=${EPICS_MODULES_PATH}/areaDetector/ADSupport
+ADCORE=${EPICS_MODULES_PATH}/areaDetector/ADCore
+"
+
+echo "$module_releases" >> ${EPICS_MODULES_PATH}/../RELEASE
+
+echo "
+EPICS_BASE=${EPICS_BASE_PATH}
+
+$module_releases
+
+ASYN=${EPICS_MODULES_PATH}/asyn
+" > RELEASE.local
+
+ln -s RELEASE.local RELEASE_PRODS.local
+ln -s RELEASE.local RELEASE_LIBS.local
+
+echo "
+BUILD_IOCS=NO
+
+WITH_BOOST=NO
+WITH_PVA=NO
+WITH_QSRV=NO
+
+WITH_BLOSC=YES
+BLOSC_EXTERNAL=NO
+
+WITH_BITSHUFFLE=NO
+WITH_GRAPHICSMAGICK=NO
+
+WITH_HDF5=YES
+HDF5_EXTERNAL=NO
+
+WITH_JSON=YES
+JSON_EXTERNAL=NO
+
+WITH_JPEG=YES
+JPEG_EXTERNAL=NO
+
+WITH_NETCDF=YES
+NETCDF_EXTERNAL=NO
+
+WITH_NEXUS=YES
+NEXUS_EXTERNAL=NO
+
+WITH_OPENCV=NO
+
+WITH_SZIP=YES
+SZIP_EXTERNAL=NO
+
+XML2_EXTERNAL=NO
```

Comment:
On your machine, when you showed me this, you were installing the XML2 package, no?

Reply:
Yes, as it is Aravis library dependency. `ADCore` by default uses the `ADSupport` version of `libxml2`, and that's why it wasn't changed there. But I had configured this before our in-person discussion.

I may change it to use the externally installed library. But before that I'd like to confirm if you agree based on some new things I've found out. Basically, `ADSupport`

- [uses version 2.9.4](https://github.com/areaDetector/ADSupport/blob/ac5d261a1abcddba11827a0b5ce500437e4485c9/supportApp/xml2Src/README.epics#L1) while Debian has packaged [version 2.9.10 for Debian 11.7](https://packages.debian.org/bullseye/libxml2) (and [2.9.14 for Debian 12](https://packages.debian.org/bookworm/libxml2)). Memory leaks and other bugs are fixed after 2.9.4, but I'm not sure how critical they are.
- patches [things for operating systems other than Linux](https://github.com/areaDetector/ADSupport/blob/master/supportApp/xml2Src/README.epics). These patches do not break anything from the original API.
- builds `libxml2` without ICU, HTML, FTP and `iconv` support, and thus removes the need to have `libicu67` in the system, whose version 67.1-7 has installation size of 33.9MB (as we checked before).
   
   [ICU support is indeed not required in Linux in `libxml` 2.9.4](https://gitlab.gnome.org/GNOME/libxml2/-/blob/v2.9.4/INSTALL.libxml2?ref_type=tags#L36) and latest versions. It has been [included as a dependency in Debian due to "longstanding issues with chromium"](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=776254), and there is already [a bug report for removing it](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=962464) due to the very same reason we are not willing to add it to our IOC containers by default.

Based on that, we could use the `ADSupport` one to avoid the cumbersome ICU depencency (also making it visible to the dynamic linker) or use the packaged version with the external installation being specified both in `BUILD_PACKAGES` and `RUNTIME_PACKAGES` by each `areaDetector` IOC as we discussed previously.

In the long-term, I see it is better to have it externally installed (considering [Debian Bug 962464](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=962464) is fixed). I'm not so sure in the sort term, though.

Reply:
Why do we need to specify it in `BUILD_PACKAGES`? The build image is huge already, so I think it could be there by default.

The 30MB difference is relevant when building the small images for static IOCs, but it`s not as relevant when we consider the size (at least for now) of the dynamic IOC images.

Furthermore, I thoroughly dislike the idea of having two versions of the same library being used in the same final executable. It`s risky and, in this case, doesn't bring a lot of benefits. I think that's the main reason I'm in favor of using the system xml2 library.

Reply:
> Why do we need to specify it in BUILD_PACKAGES? The build image is huge already, so I think it could be there by default.

My mistake. I meant including it the the `base/Dockerfile` actually. Otherwise, it won't even build `ADCore`.

>  it`s not as relevant when we consider the size (at least for now) of the dynamic IOC images.

You have a good point. I agree. Cleaning up other things may be more effective for now.

> Furthermore, I thoroughly dislike the idea of having two versions of the same library being used in the same final executable.

Yeah, having both looks messy. I was talking about having one or the other, really. But as we talked in person, having it installed in the system seems the cleaner way to have the other dependent software working, such as `libaravis`.